hid.pyx

```diff
@@ -330,6 +330,38 @@ cdef class device:
             result = hid_send_feature_report(c_hid, cbuff, c_buff_len)
         return result
 
+    def get_report_descriptor(self, int max_length=4096):
+        """Return the report descriptor up to max_length bytes.
+        If max_length is bigger than the actual descriptor, the full descriptor will be returned.
+
+        :param max_length: Maximum number of bytes to read, must be positive
+        :type max_length: int
+
+        :return:
+        :rtype: List[int]
+        :raises ValueError: If connection is not opened.
+        :raises IOError: If read error
+        """
+        if self._c_hid == NULL:
+            raise ValueError('not open')
+
+        cdef unsigned char* cbuff
+        cdef size_t c_descriptor_length = max(1, max_length)
+        cdef hid_device * c_hid = self._c_hid
+        cdef int n
+        result = []
+        try:
+            cbuff = <unsigned char *>malloc(max_length)
+            with nogil:
+                n = hid_get_report_descriptor(c_hid, cbuff, c_descriptor_length)
+            if n < 0:
+                raise IOError('read error')
+            for i in range(n):
+                result.append(cbuff[i])
```

Comment:
I'm no expert in python/cython, but this looks like the most inefficient (biggest overhead) possible way to perform such operation.
If [this](https://cython.readthedocs.io/en/latest/src/tutorial/array.html#zero-overhead-unsafe-access-to-raw-c-pointer) article is right, it should be much better alternative.

Or at least create a stack temporary variable instead of making a heap one.

Reply:
I'm no expert in Cython either, I just adapt the existing code to add this new method (see the method `get_feature_report`).

If you think there is a better way to do it, feel free to create a PR for it. My work here is done.

Reply:
@Youw yes, PR with an efficiency fix would be greatly appreciated

Reply:
So far I came up with #190.
It appears `cython.array` is not needed here, as I initially [though](https://github.com/trezor/cython-hidapi/pull/186#discussion_r1848747180), assuming `list(cbuf[:n])` does what I expect it shuld:
1) zero-overhead memory view over cython C array
2) python-implementaiton-optimized implementation of python list creation from memory view